example/synchronous_lstm/README.txt

```diff
@@ -7,32 +7,42 @@ LSTM example using Platoon *all reduce* interface
 - lstm_controller.py
 - lstm_worker.py
 - imdb.py
+It is assumed that imdb.pkl is in the same foler, otherwise it will be downloaded.
 
 
 ## HOW TO USE ##
 # USING THE LAUNCHER
 1) Assuming you are in the lstm folder.
    `cd platoon/example/lstm/`
 
-1) Launch the experiment on 2 GPUs using the platoon-launcher script.
-   `platoon-launcher lstm -D cuda0 cuda2`
+2) Launch the experiment on 2 GPUs using the platoon-launcher script.
+   `platoon-launcher lstm -D cuda0 cuda1`
 
 To see all controller parameters do: `python lstm_controller.py -h`
 To pass them via the platoon-launcher script: `platoon-launcher lstm -D cuda0 cuda2 -c=...`
 
 To see all worker parameters do: `python lstm_worker.py -h`
 To pass them via the platoon-launcher script: `platoon-launcher lstm -D cuda0 cuda2 -w=...`
 
+When the launcher is used, the outputs and errors of controller and workers are automatically
+stored in an auto-generated folder of PLATOON_LOGS/lstm/DATE_TIME.
+
+For setting THEANO_FLAGS for the workers, you can use the
+following command which sets floatX to float32 for all the workers:
+`THEANO_FLAGS=floatX=float32 platoon-launcher lstm -D cuda0 cuda1`
+
 # MANUALLY
 1) Assuming you are in the lstm folder.
    `cd platoon/example/lstm/`
 
-2) Start the controller.
-   `THEANO_FLAGS='device=cpu' python -u lstm_controller.py`
-
-3) Start the worker. Repeat as needed changing the GPU id.
-   `THEANO_FLAGS='device=gpu0' python -u lstm_worker.py`
+2) Launch the experiment. Platoon will automatically find all the available GPUs
+   and run the workers on them:
```

Comment:
Are you sure that for the manual way, platoon will detect and autaomatically launch the workers? I think this is not the goal of that  "manual" way of doing thing. I still think step 3) is needed.

Reply:
What I am sure is that the command that I wrote works and this one ('`THEANO_FLAGS='device=gpu0' python -u lstm_worker.py`') does not work. There might be other ways to run the workers as well but I am not aware of them.

Reply:
I just double checked this. When the old command is used we get an error like this:

```
usage: lstm_controller.py [-h] (--single | --multi) [-D devname [devname ...]]
                          [-w WORKER_ARGS] [--control-port CONTROL_PORT]
                          [--data-port DATA_PORT] [--data-hwm DATA_HWM]
                          [--seed SEED] [--patience PATIENCE]
                          experiment_name log_directory
lstm_controller.py: error: too few arguments
```

Which I think suggests that `experiment_name` and  `log_directory` are mandatory. So if we specify these it's going be the same as this command:
`  THEANO_FLAGS=floatX=float32 python lstm_controller.py --single lstm PATH/TO/OUTPUT`

and in this case platoon is just going to start working and by using the experiment name launches the workers. So there will be no need for step 3.